app/models/project.server.ts

```diff
@@ -128,7 +165,58 @@ export async function updateProjects({
   await db.projects.updateMany({
     where: { id: { in: ids } },
     data,
-  })
+  });
+}
+
+// edit only relatedProjects
+export async function updateRelatedProjects({
+  id,
+  data,
+}: {
+  id: string;
+  data: RelatedProjectInput;
+}) {
+  // Create related Projects
+  const createResponse = [];
+  let response;
+  for (let i = 0; i < data.relatedProjects.length; i++) {
+    let relationExist = await db.relatedProjects.count({
+      where: {
+        OR: [
+          {
+            projectAId: data.relatedProjects[i].id,
+            projectBId: id,
+          },
+          {
+            projectAId: id,
+            projectBId: data.relatedProjects[i].id,
+          },
+        ],
+      },
+    });
+
+    if (relationExist === 0) {
+      response = await db.relatedProjects.create({
+        data: {
+          projectAId: id,
+          projectBId: data.relatedProjects[i].id,
+        },
+      });
+      createResponse.push(response)
+    }
+  }
+
+  // Delete related projects
+  const relatedProjectsIds = await data.relatedProjects.map((e) => e.id);
```

Comment:
i have a doubt with this code could we set a call?

Reply:
Sure thing, want to schedule it ?